openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java

```diff
@@ -571,7 +583,8 @@ public Response createUser(
     try {
       createdUserRes = create(uriInfo, securityContext, user);
     } catch (EntityNotFoundException ex) {
-      if (securityContext.getUserPrincipal().getName().equals(create.getName())) {
+      if (isSelfSignUpEnabled
```

Comment:
we should throw a clear response back that the self signup is not enabled and UI show that clearly cc @chirag-madlani 
"Self Signup is not enabled. Please contact your admin" something on these lines

Reply:
@harshach How does this sound
"Self-registration is currently disabled. Please contact your administrator for assistance with account creation."

Reply:
**"Self Signup is not available. Please contact your Administrator."** Added this message